### PR TITLE
[RHACS] [Docs] [4.5 and later versions] ROX-21863: Setting up image mirroring for registries

### PR DESCRIPTION
**Pull request by agantony**
Version(s):
4.5+

Issue:
https://issues.redhat.com/browse/ROX-21863

Link to docs preview:
https://82567--ocpdocs-pr.netlify.app/openshift-acs/latest/operating/examine-images-for-vulnerabilities#redirecting-image-pulls-from-a-source-registry-to-a-mirrored-registry_examine-images-for-vulnerabilities
https://82567--ocpdocs-pr.netlify.app/openshift-acs/latest/operating/examine-images-for-vulnerabilities.html#scanning-images-by-using-secured-clusters_examine-images-for-vulnerabilities

SME review:
- [x] SME has approved this change.

Additional information:
- Merge into `rhacs-docs-main` and cherry-pick to:
    - `rhacs-docs-4.6`
    - `rhacs-docs-4.5`

### CODE REVIEW

File: modules/scanning-images-by-using-secured-clusters.adoc
Comment (maxwelldb) [inline modules/scanning-images-by-using-secured-clusters.adoc:9]: Consider "You can scan images by using secured clusters instead of Central services by configuring delegated image scanning."

File: modules/scanning-images-by-using-secured-clusters.adoc
Comment (maxwelldb) [inline modules/scanning-images-by-using-secured-clusters.adoc:11]: I'm not that familiar with this part of the docs. I'm assuming that capitalized things in running text like "Sensor" and "Scanner-slim" are proper product names.

Comment (maxwelldb) [inline modules/scanning-images-by-using-secured-clusters.adoc:11]: "Newly delegated" if "delegated" is a descriptor of yours. "New delegated" is fine if the feature name is "delegated registry configuration."

File: modules/enabling-seamless-image-access-and-efficient-use-of-multiple-registry-sources.adoc
Comment (maxwelldb) [inline modules/enabling-seamless-image-access-and-efficient-use-of-multiple-registry-sources.adoc:7]: Is "seamless" a crucial descriptor here? Seems sort of marketingish to me, but, again, my domain knowledge is weak.

File: modules/enabling-seamless-image-access-and-efficient-use-of-multiple-registry-sources.adoc
Comment (maxwelldb) [inline modules/enabling-seamless-image-access-and-efficient-use-of-multiple-registry-sources.adoc:37]:
```suggestion
This command creates an `ImageContentSourcePolicy` CR in the {ocp} cluster. The {ocp} CRI-O runtime environment can now pull images from `quay.io`. This configuration ensures transparent mirroring of images.
```

File: modules/scanning-images-by-using-secured-clusters.adoc
Comment (maxwelldb) [inline modules/scanning-images-by-using-secured-clusters.adoc:39]: I don't know the GUI, but is it really necessary to explain these options here? Consider deleting much of this if there is appropriate contextual guidance.

File: modules/scanning-images-by-using-secured-clusters.adoc
Comment (maxwelldb) [inline modules/scanning-images-by-using-secured-clusters.adoc:41]: Could this be raised up a level and made into an ordered list item?

File: modules/scanning-images-by-using-secured-clusters.adoc
Comment (maxwelldb) [inline modules/scanning-images-by-using-secured-clusters.adoc:52]:
```suggestion
Image integrations are now synchronized between Central and Sensor, and Sensor captures pull secrets from each namespace. Sensor then uses these credentials to authenticate to the image registries.
```

**Review by maxwelldb — COMMENTED** [review]
Left some comments and suggestions as part of peer review.

File: modules/enabling-seamless-image-access-and-efficient-use-of-multiple-registry-sources.adoc
Comment (dcaravel) [inline modules/enabling-seamless-image-access-and-efficient-use-of-multiple-registry-sources.adoc:35]: Setting up mirroring is an OCP feature (not an ACS feature) - instead of adding an example in the ACS docs should we instead refer to (or link to) the OCP docs that show how to setup? - [such as this for OCP 4.16](https://docs.redhat.com/en/documentation/openshift_container_platform/4.16/html/updating_clusters/performing-a-cluster-update#images-configuration-registry-mirror-configuring_updating-restricted-network-cluster)

File: modules/enabling-seamless-image-access-and-efficient-use-of-multiple-registry-sources.adoc
Comment (dcaravel) [inline modules/enabling-seamless-image-access-and-efficient-use-of-multiple-registry-sources.adoc:9]: Ensuring faster image pulls may not be accurate or the reason mirroring is setup.

May be best for us not to describe the benefits of mirroring (or even the setup) and leave that to the OCP docs. Instead perhaps we can keep this section focused on specifics of the ACS feature.

Such as:

> ACS supports scanning images from registry mirrors that have been configured via the OCP ImageContentSourcePolicy (ICSP), ImageDigestMirrorSet (IDMS), and/or ImageTagMirrorSet (ITMS) custom resources. (insert link to OCP docs for more info)
>
> Use delegated scanning to automatically scan images from registry mirrors (insert reference to delegated scanning section)

File: operating/examine-images-for-vulnerabilities.adoc
Comment (dcaravel) [inline operating/examine-images-for-vulnerabilities.adoc:85]: Believe this will read better if the mirroring section appears either before or after the sections describing how to setup delegated scanning (currently it splits it in the middle).  For example (assuming other PR comments addressed):

Here is the list of headings on the page:
```
About RHACS Scanner V4
Scanning images
    Understanding an...
    Supported packag...
    Supported progra...
    Supported runtim...
    Supported operat...
    Registry Mirrors      <--- **MIRRORING SECTION MOVED HERE BEFORE the delegated scanning section**
Accessing delegated image scanning
    Enhancing image ...
    Enabling seamles...   <--- CURRENT location may distract from setting up delegated scanning
    Scanning images ...
    Installing and c...
    Enabling seamles...   <--- **MIRRORING SECTION MOVED HERE AFTER delegated scanning setup described**
Setting up scanning
    Automatic scanni...
    Scanning inactiv...
About vulnerabilities
    Fetching vulnera...
    Understanding vu...
Disabling language-specific vulnerability scanning
Additional resources
```

File: modules/enabling-seamless-image-access-and-efficient-use-of-multiple-registry-sources.adoc
Comment (dcaravel) [inline modules/enabling-seamless-image-access-and-efficient-use-of-multiple-registry-sources.adoc:37]: Recommend removing this line - it is describing how OCP implements mirroring unrelated to ACS which may change in the future (and is a detail that even the OCP docs do not describe)

File: modules/enabling-seamless-image-access-and-efficient-use-of-multiple-registry-sources.adoc
Comment (dcaravel) [reply modules/enabling-seamless-image-access-and-efficient-use-of-multiple-registry-sources.adoc:7]: Perhaps just `Registry mirroring` is enough ("registry repository mirroring" is how it's referenced in [one of the headings](https://docs.redhat.com/en/documentation/openshift_container_platform/4.16/html/updating_clusters/performing-a-cluster-update#images-configuration-registry-mirror-configuring_updating-restricted-network-cluster) of the OCP 4.16 docs)

File: modules/scanning-images-by-using-secured-clusters.adoc
Comment (dcaravel) [inline modules/scanning-images-by-using-secured-clusters.adoc:11]: This may be relic from before we had the UI available, I would recommend referring to it as the `delegated scanning configuration` instead of `delegated registry configuration` to more closely resemble the UI:

image

File: modules/scanning-images-by-using-secured-clusters.adoc
Comment (dcaravel) [inline modules/scanning-images-by-using-secured-clusters.adoc:17]: With the inclusion of Scanner V4, perhaps references to `scanner-slim` in this section can be changed to just `scanner` or callout both Scanner-slim and Scanner V4 Indexer (refer to architecture page for more details)

File: modules/scanning-images-by-using-secured-clusters.adoc
Comment (dcaravel) [inline modules/scanning-images-by-using-secured-clusters.adoc:26]: Setting up image mirroring is not a pre-req for using delegated scanning, the two are not mutually exclusive. Recommend removing this line item from the pre-reqs.

Delegated scanning will scan images via the secured cluster. IF the user happens to have mirroring configured, delegated scanning will automatically attempt to scan images from the mirror references, however mirroring is NOT required to be setup to use delegated scanning, and in most user environments mirroring is NOT setup.

File: modules/scanning-images-by-using-secured-clusters.adoc
Comment (dcaravel) [reply modules/scanning-images-by-using-secured-clusters.adoc:39]: These options have confused members of ACS and users alike, the extra details were added awhile back to help explain what the different settings mean (the verbosity is also in the API docs). I recommend keeping these details.

Side note: the GUI currently has inaccurate descriptions of how the configuration works (have ROX-23464 to address that)

**Review by dcaravel — CHANGES_REQUESTED** [review]
Consider reducing the mirroring section to the suggested few lines mentioned in the comments below, and perhaps relocating the section if appropriate. We should mention all 3 supported mirroring CRs instead of just `ImageContentSourcePolicy`

File: modules/scanning-images-by-using-secured-clusters.adoc
Comment (agantony) [reply modules/scanning-images-by-using-secured-clusters.adoc:11]: Those are the major components of ACS

File: modules/scanning-images-by-using-secured-clusters.adoc
Comment (agantony) [reply modules/scanning-images-by-using-secured-clusters.adoc:41]: No, there is no particular sequence to do these steps. So I would prefer to keep it as an unordered list.

File: modules/scanning-images-by-using-secured-clusters.adoc
Comment (dcaravel) [inline modules/scanning-images-by-using-secured-clusters.adoc:21]: nit: 
```suggestion
Enabling Scanner is supported on {ocp} and Kubernetes secured clusters.
```
Or call out both "Scanner-slim" and "Scanner V4 Indexer"

File: modules/scanning-images-by-using-secured-clusters.adoc
Comment (dcaravel) [inline modules/scanning-images-by-using-secured-clusters.adoc:31]: nit: recognize this was just copied, but based on [the comment here](https://github.com/openshift/openshift-docs/pull/80974#discussion_r1747177617), the preferred way to refer to the integrated registry seems to be:

```suggestion
*** *None*: The default option. This option specifies that the secured clusters do _not_ scan any images, except for images from the integrated OpenShift image registry.
```

**Review by dcaravel — APPROVED** [review]
Looks great, few nits. Thanks for making the update!

File: modules/scanning-images-by-using-secured-clusters.adoc
Comment (kcarmichael08) [inline modules/scanning-images-by-using-secured-clusters.adoc:11]: I won't hold up merging but for the future, we try to avoid using "new" in anything except release notes, since after a few releases, it won't be new.

**Pull request merged**
